src/components/button/button.component.tsx

```diff
@@ -160,118 +161,146 @@ RenderChildrenProps) {
+      <StyledButton
+        aria-label={!children && iconType ? ariaLabel || iconType : undefined}
+        as={!disabled && href ? "a" : "button"}
+        onKeyDown={href ? handleLinkKeyDown : undefined}
+        draggable={false}
+        buttonType={buttonType}
+        disabled={disabled}
+        destructive={destructive}
+        role="button"
+        type={href ? undefined : "button"}
+        iconType={iconType}
+        size={size}
+        px={px ?? paddingX}
+        m={m}
+        noWrap={noWrap}
+        iconOnly={!!(!children && iconType)}
+        iconPosition={iconPosition}
+        target={target}
+        rel={rel}
+        fullWidth={fullWidth}
+        {...tagComponent("button", rest)}
+        {...rest}
+        {...(href && { href })}
+        ref={setRefs}
+      >
+        {renderChildren({
+          iconType,
+          iconPosition,
+          size,
+          subtext,
+          children,
+          disabled,
+          buttonType,
+          iconTooltipMessage,
+          iconTooltipPosition,
+          tooltipTarget: internalRef,
+        })}
+      </StyledButton>
+    );
   }
+);
 
-  const setRefs = useCallback(
-    (reference) => {
-      setInternalRef(reference);
-      if (!forwardRef) return;
-      if (typeof forwardRef === "object") forwardRef.current = reference;
-      if (typeof forwardRef === "function") forwardRef(reference);
-    },
-    [forwardRef]
-  );
-
-  return (
-    <StyledButton
-      aria-label={!children && iconType ? ariaLabel || iconType : undefined}
-      as={!disabled && href ? "a" : "button"}
-      onKeyDown={href ? handleLinkKeyDown : undefined}
-      draggable={false}
-      buttonType={buttonType}
-      disabled={disabled}
-      destructive={destructive}
-      role="button"
-      type={href ? undefined : "button"}
-      iconType={iconType}
-      size={size}
-      px={px ?? paddingX}
-      m={m}
-      noWrap={noWrap}
-      iconOnly={!!(!children && iconType)}
-      iconPosition={iconPosition}
-      target={target}
-      rel={rel}
-      fullWidth={fullWidth}
-      {...tagComponent("button", rest)}
-      {...rest}
-      {...(href && { href })}
-      ref={setRefs}
-    >
-      {renderChildren({
-        iconType,
-        iconPosition,
-        size,
-        subtext,
-        children,
-        disabled,
-        buttonType,
-        iconTooltipMessage,
-        iconTooltipPosition,
-        tooltipTarget: internalRef,
-      })}
-    </StyledButton>
-  );
-};
+let deprecatedButtonForwardRefWarnTriggered = false;
 
 const ButtonWithForwardRef = React.forwardRef<HTMLButtonElement, ButtonProps>(
-  (props, ref) => <Button forwardRef={ref} {...props} />
+  (props, ref) => {
+    if (!deprecatedButtonForwardRefWarnTriggered) {
+      deprecatedButtonForwardRefWarnTriggered = true;
+      Logger.deprecate(
+        "The `ButtonWithForwardRef` component is deprecated and will soon be removed. Please use a `Button` component with the `ref` prop instead."
+      );
```

Comment:
same pedantic comment as above :)

In this case I'd probably phrase it as "Please pass a ref to the `Button` component instead.". Or say "the basic Button component now supports forwardRef so please use `Button` instead".